src/connectComp.cpp

```diff
@@ -0,0 +1,265 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+#include <vector>
+#include <iostream>
+
+#include <unordered_map>
+
+#include <unordered_set>
+#include <string.h>
+#include <cassert>
+#include <cstdlib>
+
```

Comment:
`#include` your .hpp file here. You'll then notice that there is a double definition, but the below comments will fix that.